facebook/reason
# [AtomReason] Add type hint

Went straight from a JS port to a Reason port here. It's still not as
clean as I've hoped for, but we're gonna refactor the whole
MerlinService logic soon anyway.

This doesn't implement intelligent type hint expansion on selection.
This'll come later.

**Currently displays OCaml types**. I'll integrate the pretty-printed
reason types later. Just wanted to stop changing methods and submit
this.

Fixes #166.

Test plan:
- Hover over stuff in a reason file. Should display the type hint
  correctly right below.
- Hover over e.g. a comment. See the region is expanded correctly.
- Hover over a semicolon. See no error in console.
- Edit the file. Hover again. See correct behaviors.
